src/internal/incompatibility.rs

```diff
@@ -182,6 +182,14 @@ impl<P: Package, VS: VersionSet, M: Eq + Clone + Debug + Display> Incompatibilit
             return None;
         }
         let (p1, p2) = self_pkgs;
+        // We ignore self-dependencies. They are always either trivially true or trivially false,
+        // as the package version implies whether the constraint will always be fulfilled or always
+        // violated.
+        // At time of writing, the public crate API only allowed a map of dependencies,
+        // meaning it can't hit this branch, which requires two self-dependencies.
+        if p1 == p2 {
+            return None;
+        }
```

Comment:
So like, elsewhere, we'll be able to immediately convert each of these dependencies to `True` or `False`, since they're trivial -- no merging required?

Reply:
Yes, each partial solution immediately knows their value (while not filtering self-dependencies out earlier earlier keep error reporting etc. consistent)